internal/k8s/kubectl_runner.go

```diff
@@ -0,0 +1,48 @@
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/windmilleng/tilt/internal/output"
+)
+
+type kubectlRunner interface {
```

Comment:
this is fine for now, but i'm concerned that we're starting to implement a poor, half-implemented clone of kubernetes/client-go.

I wonder how long we should go down this path before using the official go client,  so that we can re-use connections, get well-typed responses and errors instead of parsing nebulous blobs, etc

Reply:
Yeah, all this jsonpathing and output parsing makes me uneasy, and using the go client sounds a lot better. I wasn't sure if we had a good reason for execing kubectl.